### DISCUSSION TITLE
Suspense is not working with server components

### DISCUSSION THREAD

**Discussion by abhinav-anshul** [vercel/next.js#49172] [App Router]
I am using the new Nextjs version13 - app directory.

I have one route as /share/slug

In the [slug] > page.tsx :
```
import React, { Suspense } from "react"
import Image from "next/image"
import Loading from "./loading"

interface Props {
  params: any
  searchParams: any
}

function Home(props: Props) {
  const { params, searchParams } = props
  const supabaseStorageBucket = process.env.NEXT_PUBLIC_SUPABASE_STORAGE_BUCKET
  const supabaseProjectUrl = process.env.NEXT_PUBLIC_SUPABASE_PROJECT_URL
  return (
    <>
      <main>
        <Suspense fallback={<Loading />}>
          <Image
            alt=""
            fill={true}
            style={{ width: "100%", height: "100%" }}
            src={`${supabaseProjectUrl}/storage/v1/object/public/${supabaseStorageBucket}/${params?.slug}.png`}
          />
        </Suspense>
      </main>
    </>
  )
}

export default Home
```

And I have defined this <Loading />component according to the new loading.tsx component located as the same level as this component.

[slug] > loading.tsx

In this file, I have a very simple loading skeleton screen. as :

```
import React from "react"
import { Skeleton } from "@/components/ui/skeleton"

function Loading() {
  return (
    <>
      <Skeleton className="w-[100px] h-[20px] rounded-full" />
    </>
  )
}

export default Loading
```

But this is not working, I don't know what is been done wrong here. I have "throttled" the network to "Slow 3g" as well, but I don't see any loading state. There is just a blank white page with no loader until the image gets loaded all of a sudden.

If anyone can help me out what I'm doing wrong?

One thing I have looked a bit into is - I maybe need to wrap my root component to the "Suspense" component. But I don't see any root component here? Do I need to wrap root layout.tsx?

**Comment by surjithctly** [comment]
Have you found any solution for this?

**Comment by abhinav-anshul** [comment]
Not really

**Comment by Sneyder2328** [comment]
actually is suspense supposed to work with loading images like that? I think for images you should use some placeholder of some sort and directly in the Image component without using suspense
